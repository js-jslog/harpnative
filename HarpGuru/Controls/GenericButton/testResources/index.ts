import { ApparatusIds, PozitionIds, PitchIds, getHarpStrata } from 'harpstrata'
import type { HarpStrata, HarpStrataProps, ActiveIds } from 'harpstrata'

const baseHarpStrataProps: HarpStrataProps = {
  apparatusId: ApparatusIds.MajorDiatonic,
  pozitionId: PozitionIds.First,
  harpKeyId: PitchIds.C,
  activeIds: [] as ActiveIds,
}

const firstPozitionHarpStrataProps: HarpStrataProps = {
  ...baseHarpStrataProps,
  pozitionId: PozitionIds.First,
}
const secondPozitionHarpStrataProps: HarpStrataProps = {
  ...baseHarpStrataProps,
  pozitionId: PozitionIds.Second,
}

const keyCHarpStrataProps: HarpStrataProps = {
  ...baseHarpStrataProps,
  harpKeyId: PitchIds.C,
}
const keyDHarpStrataProps: HarpStrataProps = {
  ...baseHarpStrataProps,
  harpKeyId: PitchIds.D,
}

export const firstPozitionHarpStrata: HarpStrata = getHarpStrata(
  firstPozitionHarpStrataProps
)
export const secondPozitionHarpStrata: HarpStrata = getHarpStrata(
  secondPozitionHarpStrataProps
)

export const keyCHarpStrata: HarpStrata = getHarpStrata(keyCHarpStrataProps)
export const keyDHarpStrata: HarpStrata = getHarpStrata(keyDHarpStrataProps)
