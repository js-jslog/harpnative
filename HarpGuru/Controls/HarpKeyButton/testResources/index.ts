import { ApparatusIds, PozitionIds, PitchIds, getHarpStrata } from 'harpstrata'
import type { HarpStrata, HarpStrataProps, ActiveIds } from 'harpstrata'

const baseHarpStrataProps: HarpStrataProps = {
  apparatusId: ApparatusIds.MajorDiatonic,
  pozitionId: PozitionIds.First,
  harpKeyId: PitchIds.C,
  activeIds: [] as ActiveIds,
}
const keyCHarpStrataProps: HarpStrataProps = {
  ...baseHarpStrataProps,
  harpKeyId: PitchIds.C,
}
const keyDHarpStrataProps: HarpStrataProps = {
  ...baseHarpStrataProps,
  harpKeyId: PitchIds.D,
}

export const keyCHarpStrata: HarpStrata = getHarpStrata(keyCHarpStrataProps)
export const keyDHarpStrata: HarpStrata = getHarpStrata(keyDHarpStrataProps)
