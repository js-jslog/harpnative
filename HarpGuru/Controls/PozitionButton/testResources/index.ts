import { ApparatusIds, PozitionIds, PitchIds, getHarpStrata } from 'harpstrata'
import type { HarpStrata, HarpStrataProps, ActiveIds } from 'harpstrata'

const baseHarpStrataProps: HarpStrataProps = {
  apparatusId: ApparatusIds.MajorDiatonic,
  pozitionId: PozitionIds.First,
  harpKeyId: PitchIds.C,
  activeIds: [] as ActiveIds,
}
const firstPozitionHarpStrataProps: HarpStrataProps = {
  ...baseHarpStrataProps,
  pozitionId: PozitionIds.First,
}
const secondPozitionHarpStrataProps: HarpStrataProps = {
  ...baseHarpStrataProps,
  pozitionId: PozitionIds.Second,
}

export const firstPozitionHarpStrata: HarpStrata = getHarpStrata(
  firstPozitionHarpStrataProps
)
export const secondPozitionHarpStrata: HarpStrata = getHarpStrata(
  secondPozitionHarpStrataProps
)
