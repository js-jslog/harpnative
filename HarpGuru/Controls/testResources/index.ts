import { getHarpStrata, ApparatusIds, PozitionIds, PitchIds } from 'harpstrata'
import type { HarpStrataProps } from 'harpstrata'

import type { HarpStrataControlProps } from '../types'

export const baseHarpStrataProps: HarpStrataProps = {
  apparatusId: ApparatusIds.MajorDiatonic,
  pozitionId: PozitionIds.First,
  harpKeyId: PitchIds.C,
  activeIds: [],
}
export const firstPozitionCMajorDiatonicHarpStrataProps: HarpStrataProps = { ...baseHarpStrataProps }
export const secondPozitionCMajorDiatonicHarpStrataProps: HarpStrataProps = { ...baseHarpStrataProps, pozitionId: PozitionIds.Second }

export const firstPozitionCMajorDiatonicHarpStrata = getHarpStrata(firstPozitionCMajorDiatonicHarpStrataProps)
export const secondPozitionCMajorDiatonicHarpStrata = getHarpStrata(secondPozitionCMajorDiatonicHarpStrataProps)

export const exampleHarpStrataControlProps: HarpStrataControlProps = {
  activeHarpStrata: firstPozitionCMajorDiatonicHarpStrata,
  setActiveHarpStrata: jest.fn(),
}
