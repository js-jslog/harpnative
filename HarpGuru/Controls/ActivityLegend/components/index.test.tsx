import React from 'react'
import { DegreeIds, PitchIds } from 'harpstrata'
import { render } from '@testing-library/react-native'

import { keyCHarpStrata } from '../../testResources'

import { ActivityLegend } from './index'

test('A component is rendered with at least a degree and a pitch', () => {
  const activeHarpStrata = keyCHarpStrata
  const setActiveHarpStrata = jest.fn()
  const { C: rootPitchId } = PitchIds
  const activeDegreeIds = [ DegreeIds.Root ]
  const activePitchIds = [ PitchIds.C ]
  const activityLegendProps = {
    activeHarpStrata, setActiveHarpStrata,
    rootPitchId, activeDegreeIds, activePitchIds
  }

  const { getByText } = render(<ActivityLegend {...activityLegendProps} />)
  expect(getByText(DegreeIds.Root)).toBeTruthy()
  expect(getByText(PitchIds.C)).toBeTruthy()
})
