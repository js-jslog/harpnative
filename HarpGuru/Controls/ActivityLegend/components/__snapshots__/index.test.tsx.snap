// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A snapshot of the legend in degree mode with some active degrees 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <View
      collapsable={false}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "paddingRight": 20,
        }
      }
    >
      <ScrollView
        data={
          Array [
            "1",
            "2b",
            "2",
            "3b",
            "3",
            "4",
            "5b",
            "5",
            "6b",
            "6",
            "7b",
            "7",
          ]
        }
        disableVirtualization={false}
        horizontal={false}
        initialNumToRender={10}
        maxToRenderPerBatch={10}
        numColumns={1}
        onEndReachedThreshold={2}
        removeClippedSubviews={false}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View
          style={null}
        >
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={0.2}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#0f9d58",
                  "height": 19.738400000000002,
                  "justifyContent": "center",
                  "width": 19.738400000000002,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#fefefe",
                    "fontSize": 12.199000000000002,
                  }
                }
              >
                1
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={null}
        >
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={0.2}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fefefe",
                  "height": 19.738400000000002,
                  "justifyContent": "center",
                  "width": 19.738400000000002,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#ddd",
                    "fontSize": 12.199000000000002,
                  }
                }
              >
                2b
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={null}
        >
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={0.2}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fefefe",
                  "height": 19.738400000000002,
                  "justifyContent": "center",
                  "width": 19.738400000000002,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#ddd",
                    "fontSize": 12.199000000000002,
                  }
                }
              >
                2
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={null}
        >
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={0.2}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fefefe",
                  "height": 19.738400000000002,
                  "justifyContent": "center",
                  "width": 19.738400000000002,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#ddd",
                    "fontSize": 12.199000000000002,
                  }
                }
              >
                3b
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={null}
        >
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={0.2}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#f4b400",
                  "height": 19.738400000000002,
                  "justifyContent": "center",
                  "width": 19.738400000000002,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#fefefe",
                    "fontSize": 12.199000000000002,
                  }
                }
              >
                3
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={null}
        >
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={0.2}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fefefe",
                  "height": 19.738400000000002,
                  "justifyContent": "center",
                  "width": 19.738400000000002,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#ddd",
                    "fontSize": 12.199000000000002,
                  }
                }
              >
                4
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={null}
        >
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={0.2}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fefefe",
                  "height": 19.738400000000002,
                  "justifyContent": "center",
                  "width": 19.738400000000002,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#ddd",
                    "fontSize": 12.199000000000002,
                  }
                }
              >
                5b
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={null}
        >
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={0.2}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#4285f4",
                  "height": 19.738400000000002,
                  "justifyContent": "center",
                  "width": 19.738400000000002,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#fefefe",
                    "fontSize": 12.199000000000002,
                  }
                }
              >
                5
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={null}
        >
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={0.2}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fefefe",
                  "height": 19.738400000000002,
                  "justifyContent": "center",
                  "width": 19.738400000000002,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#ddd",
                    "fontSize": 12.199000000000002,
                  }
                }
              >
                6b
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={null}
        >
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={0.2}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fefefe",
                  "height": 19.738400000000002,
                  "justifyContent": "center",
                  "width": 19.738400000000002,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#ddd",
                    "fontSize": 12.199000000000002,
                  }
                }
              >
                6
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={
            Object {
              "height": 0,
            }
          }
        />
      </ScrollView>
      <ScrollView
        data={
          Array [
            "C",
            "Db",
            "D",
            "Eb",
            "E",
            "F",
            "Gb",
            "G",
            "Ab",
            "A",
            "Bb",
            "B",
          ]
        }
        disableVirtualization={false}
        horizontal={false}
        initialNumToRender={10}
        maxToRenderPerBatch={10}
        numColumns={1}
        onEndReachedThreshold={2}
        removeClippedSubviews={false}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View
          style={null}
        >
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={0.2}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fefefe",
                  "height": 19.738400000000002,
                  "justifyContent": "center",
                  "width": 19.738400000000002,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#0f9d58",
                    "fontSize": 12.199000000000002,
                  }
                }
              >
                C
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={null}
        >
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={0.2}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fefefe",
                  "height": 19.738400000000002,
                  "justifyContent": "center",
                  "width": 19.738400000000002,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#ddd",
                    "fontSize": 12.199000000000002,
                  }
                }
              >
                Db
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={null}
        >
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={0.2}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fefefe",
                  "height": 19.738400000000002,
                  "justifyContent": "center",
                  "width": 19.738400000000002,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#ddd",
                    "fontSize": 12.199000000000002,
                  }
                }
              >
                D
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={null}
        >
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={0.2}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fefefe",
                  "height": 19.738400000000002,
                  "justifyContent": "center",
                  "width": 19.738400000000002,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#ddd",
                    "fontSize": 12.199000000000002,
                  }
                }
              >
                Eb
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={null}
        >
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={0.2}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fefefe",
                  "height": 19.738400000000002,
                  "justifyContent": "center",
                  "width": 19.738400000000002,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#f4b400",
                    "fontSize": 12.199000000000002,
                  }
                }
              >
                E
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={null}
        >
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={0.2}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fefefe",
                  "height": 19.738400000000002,
                  "justifyContent": "center",
                  "width": 19.738400000000002,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#ddd",
                    "fontSize": 12.199000000000002,
                  }
                }
              >
                F
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={null}
        >
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={0.2}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fefefe",
                  "height": 19.738400000000002,
                  "justifyContent": "center",
                  "width": 19.738400000000002,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#ddd",
                    "fontSize": 12.199000000000002,
                  }
                }
              >
                Gb
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={null}
        >
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={0.2}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fefefe",
                  "height": 19.738400000000002,
                  "justifyContent": "center",
                  "width": 19.738400000000002,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#4285f4",
                    "fontSize": 12.199000000000002,
                  }
                }
              >
                G
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={null}
        >
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={0.2}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fefefe",
                  "height": 19.738400000000002,
                  "justifyContent": "center",
                  "width": 19.738400000000002,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#ddd",
                    "fontSize": 12.199000000000002,
                  }
                }
              >
                Ab
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={null}
        >
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={0.2}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fefefe",
                  "height": 19.738400000000002,
                  "justifyContent": "center",
                  "width": 19.738400000000002,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#ddd",
                    "fontSize": 12.199000000000002,
                  }
                }
              >
                A
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={
            Object {
              "height": 0,
            }
          }
        />
      </ScrollView>
    </View>
    <ScrollView
      style={
        Object {
          "backgroundColor": "red",
          "width": 100,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "yellow",
              "height": 50,
              "width": 100,
            },
            Object {
              "transform": Array [
                Object {
                  "rotate": undefined,
                },
              ],
              "translateX": AnimatedValue {
                " __value": NaN,
              },
            },
          ]
        }
      />
      <Button
        title="flip"
      />
      <Button
        title="slide"
      />
    </ScrollView>
  </View>
</View>
`;

exports[`A snapshot of the legend in pitch mode with some active degrees 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <View
      collapsable={false}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "paddingRight": 20,
        }
      }
    >
      <ScrollView
        data={
          Array [
            "1",
            "2b",
            "2",
            "3b",
            "3",
            "4",
            "5b",
            "5",
            "6b",
            "6",
            "7b",
            "7",
          ]
        }
        disableVirtualization={false}
        horizontal={false}
        initialNumToRender={10}
        maxToRenderPerBatch={10}
        numColumns={1}
        onEndReachedThreshold={2}
        removeClippedSubviews={false}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View
          style={null}
        >
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={0.2}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fefefe",
                  "height": 19.738400000000002,
                  "justifyContent": "center",
                  "width": 19.738400000000002,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#0f9d58",
                    "fontSize": 12.199000000000002,
                  }
                }
              >
                1
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={null}
        >
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={0.2}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fefefe",
                  "height": 19.738400000000002,
                  "justifyContent": "center",
                  "width": 19.738400000000002,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#ddd",
                    "fontSize": 12.199000000000002,
                  }
                }
              >
                2b
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={null}
        >
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={0.2}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fefefe",
                  "height": 19.738400000000002,
                  "justifyContent": "center",
                  "width": 19.738400000000002,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#ddd",
                    "fontSize": 12.199000000000002,
                  }
                }
              >
                2
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={null}
        >
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={0.2}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fefefe",
                  "height": 19.738400000000002,
                  "justifyContent": "center",
                  "width": 19.738400000000002,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#ddd",
                    "fontSize": 12.199000000000002,
                  }
                }
              >
                3b
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={null}
        >
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={0.2}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fefefe",
                  "height": 19.738400000000002,
                  "justifyContent": "center",
                  "width": 19.738400000000002,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#f4b400",
                    "fontSize": 12.199000000000002,
                  }
                }
              >
                3
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={null}
        >
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={0.2}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fefefe",
                  "height": 19.738400000000002,
                  "justifyContent": "center",
                  "width": 19.738400000000002,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#ddd",
                    "fontSize": 12.199000000000002,
                  }
                }
              >
                4
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={null}
        >
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={0.2}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fefefe",
                  "height": 19.738400000000002,
                  "justifyContent": "center",
                  "width": 19.738400000000002,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#ddd",
                    "fontSize": 12.199000000000002,
                  }
                }
              >
                5b
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={null}
        >
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={0.2}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fefefe",
                  "height": 19.738400000000002,
                  "justifyContent": "center",
                  "width": 19.738400000000002,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#4285f4",
                    "fontSize": 12.199000000000002,
                  }
                }
              >
                5
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={null}
        >
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={0.2}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fefefe",
                  "height": 19.738400000000002,
                  "justifyContent": "center",
                  "width": 19.738400000000002,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#ddd",
                    "fontSize": 12.199000000000002,
                  }
                }
              >
                6b
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={null}
        >
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={0.2}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fefefe",
                  "height": 19.738400000000002,
                  "justifyContent": "center",
                  "width": 19.738400000000002,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#ddd",
                    "fontSize": 12.199000000000002,
                  }
                }
              >
                6
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={
            Object {
              "height": 0,
            }
          }
        />
      </ScrollView>
      <ScrollView
        data={
          Array [
            "C",
            "Db",
            "D",
            "Eb",
            "E",
            "F",
            "Gb",
            "G",
            "Ab",
            "A",
            "Bb",
            "B",
          ]
        }
        disableVirtualization={false}
        horizontal={false}
        initialNumToRender={10}
        maxToRenderPerBatch={10}
        numColumns={1}
        onEndReachedThreshold={2}
        removeClippedSubviews={false}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View
          style={null}
        >
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={0.2}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#0f9d58",
                  "height": 19.738400000000002,
                  "justifyContent": "center",
                  "width": 19.738400000000002,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#fefefe",
                    "fontSize": 12.199000000000002,
                  }
                }
              >
                C
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={null}
        >
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={0.2}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fefefe",
                  "height": 19.738400000000002,
                  "justifyContent": "center",
                  "width": 19.738400000000002,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#ddd",
                    "fontSize": 12.199000000000002,
                  }
                }
              >
                Db
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={null}
        >
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={0.2}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fefefe",
                  "height": 19.738400000000002,
                  "justifyContent": "center",
                  "width": 19.738400000000002,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#ddd",
                    "fontSize": 12.199000000000002,
                  }
                }
              >
                D
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={null}
        >
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={0.2}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fefefe",
                  "height": 19.738400000000002,
                  "justifyContent": "center",
                  "width": 19.738400000000002,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#ddd",
                    "fontSize": 12.199000000000002,
                  }
                }
              >
                Eb
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={null}
        >
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={0.2}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#f4b400",
                  "height": 19.738400000000002,
                  "justifyContent": "center",
                  "width": 19.738400000000002,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#fefefe",
                    "fontSize": 12.199000000000002,
                  }
                }
              >
                E
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={null}
        >
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={0.2}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fefefe",
                  "height": 19.738400000000002,
                  "justifyContent": "center",
                  "width": 19.738400000000002,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#ddd",
                    "fontSize": 12.199000000000002,
                  }
                }
              >
                F
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={null}
        >
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={0.2}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fefefe",
                  "height": 19.738400000000002,
                  "justifyContent": "center",
                  "width": 19.738400000000002,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#ddd",
                    "fontSize": 12.199000000000002,
                  }
                }
              >
                Gb
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={null}
        >
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={0.2}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#4285f4",
                  "height": 19.738400000000002,
                  "justifyContent": "center",
                  "width": 19.738400000000002,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#fefefe",
                    "fontSize": 12.199000000000002,
                  }
                }
              >
                G
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={null}
        >
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={0.2}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fefefe",
                  "height": 19.738400000000002,
                  "justifyContent": "center",
                  "width": 19.738400000000002,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#ddd",
                    "fontSize": 12.199000000000002,
                  }
                }
              >
                Ab
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={null}
        >
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={0.2}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fefefe",
                  "height": 19.738400000000002,
                  "justifyContent": "center",
                  "width": 19.738400000000002,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#ddd",
                    "fontSize": 12.199000000000002,
                  }
                }
              >
                A
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={
            Object {
              "height": 0,
            }
          }
        />
      </ScrollView>
    </View>
    <ScrollView
      style={
        Object {
          "backgroundColor": "red",
          "width": 100,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "yellow",
              "height": 50,
              "width": 100,
            },
            Object {
              "transform": Array [
                Object {
                  "rotate": undefined,
                },
              ],
              "translateX": AnimatedValue {
                " __value": NaN,
              },
            },
          ]
        }
      />
      <Button
        title="flip"
      />
      <Button
        title="slide"
      />
    </ScrollView>
  </View>
</View>
`;
